### PR TITLE
Replace gender_age model with fair face

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

File: facefusion/face_classifier.py [hunk 1]
```diff
@@ -54,26 +88,31 @@ def pre_check() -> bool:
...
+	crop_vision_frame = crop_vision_frame.transpose(2, 0, 1)
+	crop_vision_frame = numpy.expand_dims(crop_vision_frame, axis=0)
+	gender_id, age_id, race_id = forward(crop_vision_frame)
+	gender = categorize_gender(int(gender_id[0]))
+	age = categorize_age(int(age_id[0]))
+	race = categorize_race(int(race_id[0]))
+	return gender, age, race
+
+
+def forward(crop_vision_frame : VisionFrame) -> Tuple[Prediction, Prediction, Prediction]:
```
Comment (henryruhs) [inline facefusion/face_classifier.py:109]: use int instead of Prediction

File: facefusion/face_classifier.py [hunk 2]
```diff
@@ -4,35 +4,69 @@
...
+		return 'child'
+	elif age_id < 3:
+		return 'teen'
+	elif age_id < 7:
+		return 'adult'
+	return 'senior'
+
+
+def categorize_gender(gender_id : int) -> str:
+	if gender_id:
```
Comment (henryruhs) [inline facefusion/face_classifier.py:51]: looks like a boolean check... use == 1

File: facefusion/face_classifier.py [hunk 3]
```diff
@@ -4,35 +4,69 @@
...
+		},
+		'template': 'arcface_112_v2',
+		'size': (224, 224),
+		'mean': [ 0.485, 0.456, 0.406 ],
+		'standard_deviation': [ 0.229, 0.224, 0.225 ]
 	}
 }
 
 
+def categorize_age(age_id : int) -> str:
```
Comment (henryruhs) [inline facefusion/face_classifier.py:40]: the categories have been introduced as the gender age is not accurate enough.

we can either switch to 0-10 ranges or introduce more categories

**Review by henryruhs — CHANGES_REQUESTED** [review]

File: facefusion/face_classifier.py (hunk 1 above)
Comment (harisreedhar) [reply facefusion/face_classifier.py:79]: it is numpy array

File: facefusion/face_classifier.py (hunk 2 above)
Comment (harisreedhar) [reply facefusion/face_classifier.py]: ok

File: facefusion/face_classifier.py [hunk 4]
```diff
@@ -54,26 +58,72 @@ def pre_check() -> bool:
...
+		return 'white'
+	elif race_id == 1:
+		return 'black'
+	elif race_id == 2:
+		return 'latino'
+	elif race_id == 3 or race_id == 4:
+		return 'asian'
+	elif race_id == 5:
+		return 'indian'
+	return 'arabic'
```
Comment (henryruhs) [inline facefusion/face_classifier.py:129]: how can arabic be the default / fallback?

**Pull request merged**